web_widget_x2many_2d_matrix/static/src/js/widget_x2many_2d_matrix.js

```diff
@@ -0,0 +1,172 @@
+/* Copyright 2015 Holger Brunn <hbrunn@therp.nl>
+ * Copyright 2016 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+ * Copyright 2018 Simone Orsi <simone.orsi@camptocamp.com>
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
+
+odoo.define('web_widget_x2many_2d_matrix.widget', function (require) {
+  "use strict";
+
+  var core = require('web.core');
+  // var FieldManagerMixin = require('web.FieldManagerMixin');
+  var field_registry = require('web.field_registry');
+  var relational_fields = require('web.relational_fields');
+  var weContext = require('web_editor.context');
+  // var Helpers = require('web_widget_x2many_2d_matrix.helpers');
+  var AbstractField = require('web.AbstractField');
+  var X2Many2dMatrixRenderer = require('web_widget_x2many_2d_matrix.X2Many2dMatrixRenderer');
+
+  var WidgetX2Many2dMatrix = relational_fields.FieldOne2Many.extend({
+    widget_class: 'o_form_field_x2many_2d_matrix',
+    /**
+     * Initialize the widget & parameters.
+     *
+     * @param {Object} parent: contains the form view.
+     * @param {String} name: the name of the field.
+     * @param {Object} record: Contains the information about the database records.
+     * @param {Object} options: Contains the view options.
+     */
+    init: function (parent, name, record, options) {
+        this._super(parent, name, record, options);
+        this.init_params();
+    },
+
+    /**
+     * Initialize the widget specific parameters.
+     * Sets the axis and the values.
+     */
+    init_params: function () {
+        var node = this.attrs;
+        this.by_x_axis = {};
+        this.by_y_axis = {};
+        this.field_x_axis = node.field_x_axis || this.field_x_axis;
+        this.field_y_axis = node.field_y_axis || this.field_y_axis;
+        this.field_label_x_axis = node.field_label_x_axis || this.field_x_axis;
+        this.field_label_y_axis = node.field_label_y_axis || this.field_y_axis;
+        this.x_axis_clickable = this.parse_boolean(node.x_axis_clickable || '1');
+        this.y_axis_clickable = this.parse_boolean(node.y_axis_clickable || '1');
+        this.field_value = node.field_value || this.field_value;
+        // TODO: is this really needed? Holger?
+        for (var property in node) {
+          if (property.startsWith("field_att_")) {
+              this.fields_att[property.substring(10)] = node[property];
+          }
+        }
+        // and this?
+        this.field_editability = node.field_editability || this.field_editability;
+        this.show_row_totals = this.parse_boolean(node.show_row_totals || '1');
+        this.show_column_totals = this.parse_boolean(node.show_column_totals || '1');
+        this.init_matrix();
+    },
+    /**
+     * Initializes the Value matrix.
+     * Puts the values in the grid. If we have related items we use the display name.
+     */
+    init_matrix: function(){
+      var self = this,
```

Comment:
do i understand this correctly that this will yield the expected result even if the matrix is partial? if so, great, but please update the readme accordingly then

Reply:
yep